src/vsr/journal.zig

```diff
@@ -2083,6 +2016,13 @@ pub fn JournalType(comptime Replica: type, comptime Storage: type) type {
                 current = waiting.next;
                 waiting.next = null;
                 journal.lock_sectors(@as(*Journal.Write, @fieldParentPtr("range", waiting)));
+                // Header sector locks are always an exact match, so there's no need to re-check the
+                // the waiting writes against all other writes.
+                if (range.ring == .headers) {
+                    assert(waiting.locked);
+                    waiting.next = current;
+                    break;
+                }
```

Comment:
> Header sector locks are always an exact match

This is a good point!

Instead of `Range.offset: u64` and `Range.overlaps()`, we could store `Range.slot: Slot` -- then prepares are an exact match too -- `.overlaps` would be replaced by `.equals() { compare ring and slot }`.

Reply:
Ah, yes! This is a great point, let me try this out.

Reply:
Took a stab at it in https://github.com/tigerbeetle/tigerbeetle/pull/2273/commits/97c50edf318c863c612641d2b206c54a2c082503.

Reply:
We decided to revert this and instead just add asserts in the older version: https://github.com/tigerbeetle/tigerbeetle/pull/2273/commits/94de0abe43d6bedc4ae26e46973fb480ecdd36ec